src/package.rs

```diff
@@ -48,35 +51,47 @@ impl PackageStore {
     pub const PROTO_VENDOR_PATH: &str = "proto/vendor";
 
     /// Creates the expected directory structure for `buffrs`
-    pub async fn create() -> eyre::Result<()> {
+    pub async fn create() -> miette::Result<()> {
         let create = |dir: &'static str| async move {
-            fs::create_dir_all(dir).await.wrap_err(eyre::eyre!(
-                "Failed to create dependency folder {}",
-                Path::new(dir).canonicalize()?.to_string_lossy()
-            ))
+            fs::create_dir_all(dir)
+                .await
+                .into_diagnostic()
+                .wrap_err_with(|| miette!("failed to create {dir} directory"))
         };
 
         create(Self::PROTO_PATH).await?;
         create(Self::PROTO_VENDOR_PATH).await?;
 
         Ok(())
     }
+}
 
+impl PackageStore {
     /// Clears all packages from the file system
-    pub async fn clear() -> eyre::Result<()> {
-        fs::remove_dir_all(Self::PROTO_VENDOR_PATH)
-            .await
-            .wrap_err("Failed to uninstall dependencies")
+    pub async fn clear() -> miette::Result<()> {
+        match fs::remove_dir_all(Self::PROTO_VENDOR_PATH).await {
+            Ok(()) => Ok(()),
+            Err(err) if matches!(err.kind(), std::io::ErrorKind::NotFound) => {
+                Err(miette!("directory {} not found", Self::PROTO_VENDOR_PATH))
+            }
+            Err(_) => Err(miette!(
+                "failed to clear {} directory",
+                Self::PROTO_VENDOR_PATH,
+            )),
+        }
     }
+}
 
+impl PackageStore {
     /// Unpacks a package into a local directory
-    pub async fn unpack(package: &Package) -> eyre::Result<()> {
+    pub async fn unpack(package: &Package) -> miette::Result<()> {
         let mut tar = Vec::new();
 
         let mut gz = flate2::read::GzDecoder::new(package.tgz.clone().reader());
 
         gz.read_to_end(&mut tar)
-            .wrap_err("Failed to decompress package")?;
+            .into_diagnostic()
+            .wrap_err_with(|| miette!("failed to decompress package {}", package.name()))?;
```

Comment:
i would use the short syntax everywhere, we dont have performance problems at all, imo readability is more important

Reply:
I think you actually suggested some of these, but I can revert.